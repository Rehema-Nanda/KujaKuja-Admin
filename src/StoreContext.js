// https://mobx-react.js.org/recipes-context

import React from "react";
import PropTypes from "prop-types";
import RootStore from "./stores/RootStore";

export const StoreContext = React.createContext(null);

export const StoreProvider = ({ children }) => {
    const store = new RootStore();
    return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};
StoreProvider.propTypes = {
    children: PropTypes.node.isRequired,
};

export const StoreConsumer = StoreContext.Consumer;

export const useStore = () => {
    const store = React.useContext(StoreContext);
    if (!store) {
        // this is especially useful in TypeScript so you don't need to be checking for null all the time
        throw new Error("useStore must be used within a StoreProvider.");
    }
    return store;
};
