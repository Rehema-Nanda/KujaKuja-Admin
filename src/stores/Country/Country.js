import {
    decorate, computed, observable, comparer,
} from "mobx";
import lineToPolygon from "@turf/line-to-polygon";

class Country {
    store = null;
    id = null;
    enabled = null;
    name = null;
    isoTwoLetterCode = null;
    geojson = null;
    lat = null;
    lng = null;
    createdAt = null;
    updatedAt = null;

    constructor(store, json = null) {
        this.store = store;

        if (json) {
            this.id = json.id;
            this.updateFromJson(json);
        }
    }

    get asJson() {
        return {
            enabled: this.enabled,
            name: this.name,
            iso_two_letter_code: this.isoTwoLetterCode,
            geojson: JSON.stringify(this.geojson),
            lat: this.lat,
            lng: this.lng,
            created_at: this.createdAt,
            updated_at: this.updatedAt,
        };
    }

    updateFromJson(json) {
        this.enabled = json.enabled;
        this.name = json.name;
        this.isoTwoLetterCode = json.iso_two_letter_code;
        this.geojson = json.geojson;
        this.lat = json.lat;
        this.lng = json.lng;
        this.createdAt = json.created_at;
        this.updatedAt = json.updated_at;

        this.fixGeoJson();
    }

    fixGeoJson() {
        // TODO: https://trello.com/c/wxv8WY0m/426-frontend-and-admin-backend-are-both-doing-linestring-to-polygon-conversion-but-differently-json-parsing
        if (this.geojson && this.geojson.geometry.type === "LineString") {
            this.geojson = lineToPolygon(this.geojson);
        }
    }

    async delete() {
        if (!this.id) {
            return;
        }
        await this.store.apiService.deleteCountry(this.id);
        await this.store.removeCountry(this);
    }

    async save() {
        const json = this.asJson;
        if (this.id) {
            await this.store.apiService.updateCountry(this.id, json);
        }
        else {
            const responseData = await this.store.apiService.createCountry(json);
            this.id = responseData.id;
            await this.store.addCountry(this);
        }
    }
}
decorate(Country, {
    lat: observable,
    lng: observable,
    asJson: computed({ equals: comparer.shallow }),
});

export default Country;
