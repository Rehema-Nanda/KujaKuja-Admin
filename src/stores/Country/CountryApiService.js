import ApiService from "../ApiService";

export default class CountryApiService extends ApiService {
    fetchAllCountries = async (config) => {
        const countriesResponse = await this.makeHttpCall("countries/paginator", "post", config);
        return countriesResponse.data; // TODO: change data structure returned by API
    };

    fetchEnabledCountries = async () => {
        const enabledCountriesResponse = await this.makeHttpCall("countries/enabled");
        return enabledCountriesResponse.data.data; // TODO: change data structure returned by API
    };

    fetchCountryById = async (id) => {
        const countryResponse = await this.makeHttpCall(`countries/${id}`);
        return countryResponse.data.data; // TODO: change data structure returned by API
    };

    deleteCountry = async (id) => {
        const deleteCountryResponse = await this.makeHttpCall(`countries/${id}`, "delete");
        // TODO: POST/PUT/DELETE endpoints should return string ID to be consistent with GET
        deleteCountryResponse.data.id = deleteCountryResponse.data.id.toString();
        return deleteCountryResponse.data;
    };

    createCountry = async (countryJson) => {
        const config = { data: countryJson };
        const createCountryResponse = await this.makeHttpCall("countries", "post", config);
        // TODO: POST/PUT/DELETE endpoints should return string ID to be consistent with GET
        createCountryResponse.data.id = createCountryResponse.data.id.toString();
        return createCountryResponse.data;
    };

    updateCountry = async (id, countryJson) => {
        const config = { data: countryJson };
        const updateCountryResponse = await this.makeHttpCall(`countries/${id}`, "put", config);
        // TODO: POST/PUT/DELETE endpoints should return string ID to be consistent with GET
        updateCountryResponse.data.id = updateCountryResponse.data.id.toString();
        return updateCountryResponse.data;
    };
}