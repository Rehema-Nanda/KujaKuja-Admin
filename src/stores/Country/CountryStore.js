import {
    flow, decorate, observable, computed,
} from "mobx";
import Country from "./Country";
import CountryApiService from "./CountryApiService";
import AppConfig from "../../AppConfig";

class CountryStore {
    countries = [];
    count = 0;
    enabledCountries = [];
    isLoading = false;

    constructor(rootStore) {
        this.rootStore = rootStore;
        this.apiService = new CountryApiService(AppConfig.API_URL);

        this.loadCountries();
        this.loadEnabledCountries();
    }

    get countriesSortedByName() {
        return this.countries.slice().sort((a, b) => {
            if (a.name < b.name) {
                return -1;
            }
            else if (a.name > b.name) {
                return 1;
            }
            return 0;
        });
    }

    loadCountries = flow(function* (config) { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const countriesJson = yield this.apiService.fetchAllCountries(config);
            this.countries.replace(countriesJson.data.map((countryJson) => {
                return new Country(this, countryJson);
            }));
            this.count = countriesJson.count;
            this.isLoading = false;
        }
        catch (error) {
            console.error(`CountryStore failed to load countries. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    loadEnabledCountries = flow(function* () { // eslint-disable-line func-names
        this.isLoading = true;
        try {
            const countriesJson = yield this.apiService.fetchEnabledCountries();
            this.enabledCountries.replace(countriesJson.map((countryJson) => {
                return new Country(this, countryJson);
            }));
            this.isLoading = false;
        }
        catch (error) {
            console.error(`CountryStore failed to load enabled countries. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    updateCountryFromServer = flow(function* (id) { // eslint-disable-line func-names
        this.isLoading = true;
        const countriesJson = yield this.apiService.fetchCountryById(id);
        let country = this.countries.find((c) => {
            return c.id === id;
        });

        if (countriesJson.length === 0 && country) {
            // country no longer exists on the server, remove it locally
            yield this.removeCountry(country);
            country = null;
        }
        if (countriesJson.length > 0) {
            if (country) {
                country.updateFromJson(countriesJson[0]);
            }
            else {
                country = new Country(this, countriesJson[0]);
                this.countries.push(country);
            }
        }
        this.isLoading = false;
        return country;
    });

    removeCountry = flow(function* (country) { // eslint-disable-line func-names, require-yield
        this.countries.splice(this.countries.indexOf(country), 1);
    });

    addCountry = flow(function* (country) { // eslint-disable-line func-names, require-yield
        this.countries.push(country);
    });
}
decorate(CountryStore, {
    countries: observable,
    enabledCountries: observable,
    countriesSortedByName: computed,
    isLoading: observable,
});

export default CountryStore;
