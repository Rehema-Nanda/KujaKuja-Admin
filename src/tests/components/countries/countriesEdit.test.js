import React from "react";
import { mount } from "enzyme";
import CountriesEdit from "../../../components/countries/edit";
import { StoreProvider } from "../../../StoreContext";
import CountryStore from "../../../stores/Country/CountryStore";
import Country from "../../../stores/Country/Country";

jest.mock("../../../stores/Country/CountryStore");

const countries = [
    new Country(
        null,
        {
            id: "1",
            enabled: false,
            name: "Afghanistan",
            iso_two_letter_code: "AF",
            geojson: null,
            lat: "0.00000000",
            lng: "0.00000000",
            created_at: null,
            updated_at: null,
        },
    ),
];

const mockCountryDelete = jest.fn();
countries[0].delete = mockCountryDelete;

const mockUpdateCountryFromServer = jest.fn(() => {
    return countries[0];
});
CountryStore.mockImplementation(() => {
    return {
        countries: countries,
        updateCountryFromServer: mockUpdateCountryFromServer,
    };
});

describe("CountriesEdit component", () => {
    const testProps = {
        history: {
            push: jest.fn(),
        },
        match: {
            params: {
                id: 1,
            },
        },
    };
    const wrapper = mount(<CountriesEdit {...testProps} />, {
        wrappingComponent: StoreProvider,
    });

    it("should render correctly", () => {
        expect(mockUpdateCountryFromServer).toHaveBeenCalled();
        // console.log(wrapper.instance().simpleState.country);
        wrapper.update();
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should push history when cancelForm function is called", () => {
        const e = { preventDefault: jest.fn() };
        const cancelForm = jest.spyOn(wrapper.instance(), "cancelForm");
        cancelForm(e);
        expect(testProps.history.push).toHaveBeenCalledWith("/countries");
    });

    it("should change modal state when toggleDeleteConfirmationModal function is called", () => {
        const modalState = wrapper.instance().simpleState.isDeleteConfirmationModalOpen;
        const toggleDeleteConfirmationModal = jest.spyOn(wrapper.instance(), "toggleDeleteConfirmationModal");
        toggleDeleteConfirmationModal();
        expect(wrapper.instance().simpleState.isDeleteConfirmationModalOpen).toEqual(!modalState);
    });

    it("should call delete function on Country object and push history when delete function is called", async () => {
        const deleteFn = jest.spyOn(wrapper.instance(), "delete");
        await deleteFn();
        expect(mockCountryDelete).toHaveBeenCalled();
        expect(testProps.history.push).toHaveBeenCalledWith("/countries");
    });
});
