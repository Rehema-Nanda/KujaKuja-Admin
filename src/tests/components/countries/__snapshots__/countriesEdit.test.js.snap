// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountriesEdit component should render correctly 1`] = `
"<CountriesEdit history={{...}} match={{...}}>
  <div className=\\"page\\">
    <div className=\\"title-background-parent\\">
      <div className=\\"title-background-shape\\" />
    </div>
    <Banner pageTitle=\\"country\\" elementAction=\\"update\\" buttonFunction={[Function]} buttonDisabled={false}>
      <Container tag=\\"div\\">
        <div className=\\"container\\">
          <Row tag=\\"div\\">
            <div className=\\"row\\">
              <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
                <div className=\\"col-sm-7 col-md-8\\">
                  <div className=\\"page-title\\">
                    <h1 className=\\"left\\">
                      Country
                    </h1>
                    <p>
                      Manage your
                       
                      country
                    </p>
                  </div>
                </div>
              </Col>
              <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
                <div className=\\"col-sm-5 col-md-4\\">
                  <div className=\\"page-title\\" />
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </Container>
    </Banner>
    <ConfirmationModal buttonFunction={[Function]} toggle={[Function]} isOpen={false}>
      <Modal isOpen={false} toggle={[Function]} autoFocus={true} centered={false} role=\\"dialog\\" backdrop={true} keyboard={true} zIndex={1050} fade={true} onOpened={[Function: noop]} onClosed={[Function: noop]} modalTransition={{...}} backdropTransition={{...}} />
    </ConfirmationModal>
    <Container tag=\\"div\\">
      <div className=\\"container\\">
        <div className=\\"white-page-bg\\">
          <Row tag=\\"div\\">
            <div className=\\"row\\">
              <Col lg={{...}} tag=\\"div\\" widths={{...}}>
                <div className=\\"col-lg-10 offset-lg-1\\">
                  <div>
                    <AvForm onValidSubmit={[Function]} tag={[Function: Form]} model={{...}} validationEvent={{...}} method=\\"get\\" onSubmit={[Function: onSubmit]} onKeyDown={[Function: onKeyDown]} onInvalidSubmit={[Function: onInvalidSubmit]}>
                      <Form noValidate={true} action=\\"#\\" method=\\"get\\" onSubmit={[Function]} onKeyDown={[Function: onKeyDown]} className=\\"av-valid\\" tag=\\"form\\">
                        <form noValidate={true} action=\\"#\\" method=\\"get\\" onSubmit={[Function]} onKeyDown={[Function: onKeyDown]} className=\\"av-valid\\">
                          <Row tag=\\"div\\">
                            <div className=\\"row\\">
                              <Col tag=\\"div\\" widths={{...}}>
                                <div className=\\"col\\">
                                  <AvField name=\\"name\\" label=\\"Name\\" required={true} value=\\"Afghanistan\\">
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"name\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"name\\" className=\\"\\">
                                              Name
                                            </label>
                                          </Label>
                                          <AvInput id=\\"name\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"name\\" required={true} value=\\"Afghanistan\\" validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"name\\" required={true} value=\\"Afghanistan\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"name\\" type=\\"text\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"name\\" required={true} value=\\"Afghanistan\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} id=\\"name\\" type=\\"text\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                  <AvField name=\\"iso_two_letter_code\\" label=\\"ISO Two-Letter Code\\" required={true} maxLength=\\"2\\" minLength=\\"2\\" value=\\"AF\\">
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"iso_two_letter_code\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"iso_two_letter_code\\" className=\\"\\">
                                              ISO Two-Letter Code
                                            </label>
                                          </Label>
                                          <AvInput id=\\"iso_two_letter_code\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"iso_two_letter_code\\" required={true} maxLength=\\"2\\" minLength=\\"2\\" value=\\"AF\\" validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"iso_two_letter_code\\" required={true} maxLength=\\"2\\" minLength=\\"2\\" value=\\"AF\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"iso_two_letter_code\\" type=\\"text\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"iso_two_letter_code\\" required={true} maxLength=\\"2\\" minLength=\\"2\\" value=\\"AF\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} id=\\"iso_two_letter_code\\" type=\\"text\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                  <AvField type=\\"checkbox\\" name=\\"enabled\\" label=\\"Enabled\\" checked={false}>
                                    <AvGroup check={true} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={true} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-check\\">
                                          <AvInput id=\\"enabled\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"checkbox\\" name=\\"enabled\\" checked={false} validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"checkbox\\" name=\\"enabled\\" checked={false} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} value={false} className=\\"is-untouched is-pristine av-valid\\" id=\\"enabled\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"enabled\\" checked={false} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} value={false} id=\\"enabled\\" type=\\"checkbox\\" className=\\"is-untouched is-pristine av-valid form-check-input\\" />
                                            </Input>
                                          </AvInput>
                                          <Label for=\\"enabled\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"enabled\\" className=\\"\\">
                                              Enabled
                                            </label>
                                          </Label>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                  <AvField type=\\"number\\" name=\\"lat\\" label=\\"Latitude\\" required={true} validate={{...}} value=\\"0.00000000\\" onChange={[Function]}>
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"lat\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"lat\\" className=\\"\\">
                                              Latitude
                                            </label>
                                          </Label>
                                          <AvInput id=\\"lat\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"number\\" name=\\"lat\\" required={true} validate={{...}} value=\\"0.00000000\\" onChange={[Function]} validationEvent=\\"\\" trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"number\\" name=\\"lat\\" required={true} value=\\"0.00000000\\" onChange={[Function: bound onChangeHandler]} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"lat\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"lat\\" required={true} value=\\"0.00000000\\" onChange={[Function: bound onChangeHandler]} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} id=\\"lat\\" type=\\"number\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                  <AvField type=\\"number\\" name=\\"lng\\" label=\\"Longitude\\" required={true} validate={{...}} value=\\"0.00000000\\" onChange={[Function]}>
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"lng\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"lng\\" className=\\"\\">
                                              Longitude
                                            </label>
                                          </Label>
                                          <AvInput id=\\"lng\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"number\\" name=\\"lng\\" required={true} validate={{...}} value=\\"0.00000000\\" onChange={[Function]} validationEvent=\\"\\" trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} type=\\"number\\" name=\\"lng\\" required={true} value=\\"0.00000000\\" onChange={[Function: bound onChangeHandler]} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"lng\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"lng\\" required={true} value=\\"0.00000000\\" onChange={[Function: bound onChangeHandler]} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} id=\\"lng\\" type=\\"number\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                </div>
                              </Col>
                              <Col tag=\\"div\\" widths={{...}}>
                                <div className=\\"col\\">
                                  <p className=\\"small\\">
                                    To add a center point or outline, click one of the tool buttons and then the map. Click a blue shape to make it editable (turns yellow). Click away from a yellow shape to set it (turns blue).
                                  </p>
                                  <p className=\\"small\\">
                                    Make sure to move the center point to within the bounds of the country when enabling a country for the first time.
                                  </p>
                                  <div className=\\"data-map\\">
                                    <div />
                                  </div>
                                </div>
                              </Col>
                            </div>
                          </Row>
                          <BtnRow buttonLabel=\\"Update\\" cancelForm={[Function]}>
                            <Row className=\\"btn-row\\" tag=\\"div\\">
                              <div className=\\"btn-row row\\">
                                <Col tag=\\"div\\" widths={{...}}>
                                  <div className=\\"col\\" />
                                </Col>
                                <Col tag=\\"div\\" widths={{...}}>
                                  <div className=\\"col\\">
                                    <Button color=\\"primary\\" className=\\"primary cancel\\" onClick={[Function]} tag=\\"button\\">
                                      <button type=\\"button\\" onClick={[Function: bound onClick]} className=\\"primary cancel btn btn-primary\\" aria-label={{...}}>
                                        Back
                                      </button>
                                    </Button>
                                  </div>
                                </Col>
                              </div>
                            </Row>
                          </BtnRow>
                          <p className=\\"error-message\\" style={{...}} />
                        </form>
                      </Form>
                    </AvForm>
                  </div>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </div>
    </Container>
  </div>
</CountriesEdit>"
`;
